proto/proto/store/rpc.proto

```diff
@@ -422,3 +425,35 @@ message PaginationInfo {
     // starting from the next block to this one (ie, request.block_range.block_from = block_num + 1).
     fixed32 block_num = 2;
 }
+
+// SYNC TRANSACTIONS
+// ================================================================================================
+
+// Transactions synchronization request.
+//
+// Allows clients to sync transactions for specific accounts within a block range.
+message SyncTransactionsRequest {
+    // Block range from which to start synchronizing.
+    BlockRange block_range = 1;
+
+    // Accounts to sync transactions for.
+    repeated account.AccountId account_ids = 2;
+}
+
+// Represents the result of syncing transactions request.
+message SyncTransactionsResponse {
+    // Pagination information.
+    PaginationInfo pagination_info = 1;
+
+    // List of transaction records.
+    repeated TransactionRecord transaction_records = 2;
+}
+
+// Represents a transaction record.
+message TransactionRecord {
+    // Block number in which the transaction was executed.
+    fixed32 block_num = 1;
+
+    // A transaction header.
+    transaction.TransactionHeader transaction_header = 2;
```

Comment:
nit: maybe I'd change this to just `header`. Let's add this to the follow-ups as well.